src/ResourceManager/Compute/Commands.Compute/Strategies/ComputeRp/ImageEx.cs

```diff
@@ -25,14 +25,78 @@ namespace Microsoft.Azure.Commands.Compute.Strategies.ComputeRp
 {
     static class ImageEx
     {
-        public static int[] UpdatePorts(this OperatingSystemTypes osType, int[] ports)
-            => ports ?? (osType == OperatingSystemTypes.Windows ? new[] { 3389, 5985 } : new[] { 22 });
+        public static int[] UpdatePorts(this ImageAndOsType imageAndOsType, int[] ports)
+            => ports ?? imageAndOsType?.OsType.CreatePorts();
+
+        private static int[] CreatePorts(this OperatingSystemTypes osType)
+            => osType == OperatingSystemTypes.Windows ? new[] { 3389, 5985 }
+                : osType == OperatingSystemTypes.Linux ? new[] { 22 }
+                : null;
 
         public static async Task<ImageAndOsType> UpdateImageAndOsTypeAsync(
-            this IClient client, string imageName, string location)
+            this IClient client,
+            ImageAndOsType imageAndOsType,
+            string resourceGroupName,
+            string imageName,
+            string location)
         {
-            if (imageName.Contains(':'))
+            if (imageAndOsType != null)
+            {
+                return imageAndOsType;
+            }
+
+            if (imageName.Contains("/"))
             {
+                var imageArray = imageName.Split('/');
+                if (imageArray.Length != 9)
+                {
+                    throw new ArgumentException("Invalid image resource id  '" + imageName + "'.");
+                }
+                // has to be ""
```

Comment:
I would move image resource id parsing into a helper method, or better yet, use this common code, and simply ensure that the properties are what you expect:  https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Utilities/Models/ResourceIdentifier.cs

This will give you subscription, resourceGroupName, resourceName and resourceType (Microsoft.Comute/images)